packages/core/test/setup.types.test.ts

```diff
@@ -2184,9 +2184,48 @@ describe('setup()', () => {
 
     const actor = createActor(machine);
 
-    actor.getSnapshot().getMeta().a satisfies { layout: string } | undefined;
+    actor.getSnapshot().getMeta()['(machine).a'] satisfies
```

Comment:
it looks like this revealed some issue with the `StateId` type

Reply:
See [520a829](https://github.com/statelyai/xstate/pull/4979/commits/520a829fb6c8180cc9b82a8015216886dd00c2d2)

Reply:
And that's why we should not write huge type tests like "should support typing meta properties". Every "assertion" contained in it should be a separate test to avoid issues like this

Reply:
I'll separate this into multiple tests

Reply:
Separated: [e337836](https://github.com/statelyai/xstate/pull/4979/commits/e3378365c6d04fe2ea7b40520320b741da7c4e8a)